analyzer/restapi-plugin/src/main/java/eu/fasten/analyzer/restapiplugin/server/OpenAPIServer.java

```diff
@@ -0,0 +1,418 @@
+                                    String pkgName = params.pathParameter("pkg_name").getString();
+                                    long ts = params.pathParameter("timestamp").getInteger();
+                                    boolean transitive = params.queryParameter("transitive").getBoolean();
+                                    // TODO: check the transitive value and define it as false by default.
+                                    final var timestamp = this.getProperTimestamp(ts);
+
+                                    logger.debug("Parsed parameters converted to string (" + forge + ", "
+                                            + pkgName + ", " + ts + ").");
+
+                                    dslContext.connection(connection -> {
+                                        metadataDao.setContext(DSL.using(connection));
+                                        try {
+                                            String queryResult = metadataDao.rebuildDependencyNet(forge, pkgName, timestamp, transitive);
+                                            logger.debug("Got the reply: " + queryResult.substring(0, 50) + "...");
+
+                                            if (!queryResult.isEmpty())
+                                                routingContext.response().setStatusCode(200)
+                                                        .putHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+                                                        .end(queryResult);
+                                            else
+                                                routingContext.fail(404, new Exception("Query not found!"));
+                                            // TODO: nice to have more specific exceptions ex: pkg not found, pkgName not found etc.
+                                        } catch (RuntimeException e) {
+                                            routingContext.fail(404, new Exception("No results found for this query!"));
+                                            logger.error("Error querying the database: ", e);
+                                            throw e;
+                                        }
+                                    });
+                                });
+
+                /**
+                 * "/api/{forge}/{pkg_name}/cg/{timestamp}"
+                 *
+                 * Given a package name and a timestamp, retrieve its call graph
+                 * Use case: A user wants to run a custom analysis locally.
+                 *
+                 * Return: A JSON-serialized RevisionCallGraph
+                 *
+                 * REST examples:
+                 *  GET /api/mvn/org.slf4j:slf4j-api/cg/1233323123
+                 *  GET /api/pypi/numpy/cg/1233323123?transitive=true
+                 *  GET /api/mvn/au.org.consumerdatastandards:reflection/cg/1580706000
+                 */
+                routerFactory.addHandlerByOperationId
```

Comment:
It would be nice to have these handlers in separate methods. Having a method of 350 lines long is not very nice